crates/youki/src/commands/run.rs

```diff
@@ -4,16 +4,22 @@ use anyhow::{Context, Result};
 use libcontainer::{container::builder::ContainerBuilder, syscall::syscall::create_syscall};
 use liboci_cli::Run;
 
+use crate::workload::executor::default_executors;
+
 pub fn run(args: Run, root_path: PathBuf, systemd_cgroup: bool) -> Result<()> {
     let syscall = create_syscall();
-    let mut container = ContainerBuilder::new(args.container_id.clone(), syscall.as_ref())
-        .with_pid_file(args.pid_file.as_ref())?
-        .with_console_socket(args.console_socket.as_ref())
-        .with_root_path(root_path)?
-        .with_preserved_fds(args.preserve_fds)
-        .as_init(&args.bundle)
-        .with_systemd(systemd_cgroup)
-        .build()?;
+    let mut container = ContainerBuilder::new(
+        args.container_id.clone(),
+        syscall.as_ref(),
+        default_executors(),
```

Comment:
I would simplify this. The idea behind the `new` method is that we specify all the arguments that are required and then use the `with` methods on the builder to customize optional behavior. A consumer of the API should not know what this executor thing is unless they specifically seek out to change them. So remove the executors from new and add a `with_executors` method that would replace the default executor with the specified executors.